### ISSUE TITLE
Aquarius fails to process MetadataCreated event when creating dataset using Barge

### ISSUE THREAD

**Issue by rogargon** [oceanprotocol/aquarius#847]
**Describe the bug**

Trying to use a market on top of a locally deployed Barge. Capable of going through the process of publishing a new dataset. Everything seems to work a success is reported after signing the final 3 transactions. However, the just published cannot be found.

Looking at Aquarius output, the creation event is captures but it seems to be an error when calling decrypt on the provider:

aquarius.events.events_monitor - ERROR - Error processing MetadataCreated event: HTTPConnectionPool(host='0.0.0.0', port=8030): Max retries exceeded with url: /api/services/decrypt (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f715cd0da60>: Failed to establish a new connection: [Errno 111] Connection refused'))

event=AttributeDict({'logIndex': 0, 'transactionIndex': 0, 'transactionHash': HexBytes('0xe90b972eb6d68b552b988b2aed94485cbf2803b7e17e58bb5f752089dabd775e'), 'blockHash': HexBytes('0xd36cd..., 'type': 'mined', 'removed': False})

**To Reproduce**
1. Start Barge locally
2. Copy local contracts to Market
3. Start market
4. Go through new dataset publishing process, which finishes successfully on the user interface after completing the 3 required transactions with MetaMask
5. The new dataset cannot be found, even after some minutes, the start_barge.sh command output shows the error detailed above, it seems Aquarius failed to process the creation event.

**Expected behavior**
The just created dataset appears in the market's user interface.

**Additional context**
Using the last version of Barge, with Aquarius v4.1.0.

**Comment by rogargon** [comment]
It might be related to #796, though a local infrastructure provided by Barge is being used in this case.

**Comment by calina-c** [comment]
Do you have a custom Aquarius? I see it's trying to connect to a provider with `host='0.0.0.0', port=8030`, but that is not the correct provider address in barge.

**Comment by rogargon** [comment]
No, the whole infrastructure is started using Barge. Barge´s docker-compose file for the provider publishes its port as 8030 on the host:

```

version: '3'
services:
  provider:
    image: oceanprotocol/provider-py:${PROVIDER_VERSION:-latest}
    ports:
      - 8030:8030
    networks:
      backend:
        ipv4_address: 172.15.0.4
    depends_on:
    - ocean-contracts
...
```

There is where it can be found from the browser. The market doesn't have access to the backend network created by Barge.

**Comment by rogargon** [comment]
The endpoint http://172.15.0.4:8030 doesn't validate using the market form. That IP is not visible from the browser when using the market as it is an internal Docker network. However, the provider port is published on host port 8030. There, it is visible and responds to the verification request from the market using 0.0.0.0:8030. 

The problem might be that later Aquarius is trying to reach the provider at 0.0.0.0:8030 and it is not visible from the Aquarius container. That might be why the decrypt call fails...
